diff --git a/Transfering b/Transfering
new file mode 100755
index 0000000..f745783
--- /dev/null
+++ b/Transfering
@@ -0,0 +1 @@
+mvn exec:java -Dexec.mainClass=com.paulojeronimo.web3j.Transfering -Dexec.args="$wallet web3jtest"
diff --git a/src/main/java/com/paulojeronimo/web3j/Transfering.java b/src/main/java/com/paulojeronimo/web3j/Transfering.java
new file mode 100644
index 0000000..666c0f0
--- /dev/null
+++ b/src/main/java/com/paulojeronimo/web3j/Transfering.java
@@ -0,0 +1,33 @@
+package com.paulojeronimo.web3j;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.ExecutionException;
+
+import org.web3j.crypto.CipherException;
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.WalletUtils;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.protocol.exceptions.TransactionTimeoutException;
+import org.web3j.protocol.http.HttpService;
+import org.web3j.tx.Transfer;
+import org.web3j.utils.Convert;
+
+public class Transfering {
+    public static void main(String[] args) throws IOException, 
+            CipherException, InterruptedException,
+            ExecutionException, TransactionTimeoutException {
+
+        Web3j web3 = Web3j.build(new HttpService());
+        String walletSource = args[0];
+        Credentials credentials = WalletUtils.loadCredentials(args[1], walletSource);
+        String myWalletAddr = "0x" + Util.walletId(walletSource);
+        TransactionReceipt transactionReceipt = Transfer.sendFundsAsync(web3, credentials, myWalletAddr,
+                BigDecimal.valueOf(0.01), Convert.Unit.ETHER).get();
+        System.out.println("Funds transfer completed...");
+        System.out.println("Hash: " + transactionReceipt.getBlockHash());
+        System.out.println("Block number: " + transactionReceipt.getBlockNumber());
+
+    }
+}
diff --git a/src/main/java/com/paulojeronimo/web3j/Util.java b/src/main/java/com/paulojeronimo/web3j/Util.java
new file mode 100644
index 0000000..27501df
--- /dev/null
+++ b/src/main/java/com/paulojeronimo/web3j/Util.java
@@ -0,0 +1,14 @@
+package com.paulojeronimo.web3j;
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+public class Util {
+    public static String walletId(final String walletSource) {
+        Pattern p = Pattern.compile(".*--(.*)\\.json");
+        Matcher m = p.matcher(walletSource);
+        if (m.find())
+            return m.group(1);
+        throw new RuntimeException("walletId not found!");
+    }
+}
